src/util/intOps.ml

```diff
@@ -202,59 +202,59 @@ struct
   let to_int64 x = x
   let of_string = Int64.of_string
   let to_string = Int64.to_string
-  let of_bigint = Big_int_Z.int64_of_big_int
-  let to_bigint = Big_int_Z.big_int_of_int64
+  let of_bigint = Z.to_int64
+  let to_bigint = Z.of_int64
 end
 
-module BigIntOpsBase : IntOpsBase with type t = Big_int_Z.big_int =
+module BigIntOpsBase : IntOpsBase with type t = Z.t =
 struct
-  type t = Big_int_Z.big_int
-  let zero = Big_int_Z.zero_big_int
-  let one = Big_int_Z.unit_big_int
+  type t = Z.t
+  let zero = Z.zero
+  let one = Z.one
   let upper_bound = None
   let lower_bound = None
 
-  let neg = Big_int_Z.minus_big_int
-  let abs = Big_int_Z.abs_big_int
-  let add = Big_int_Z.add_big_int
-  let sub = Big_int_Z.sub_big_int
-  let mul = Big_int_Z.mult_big_int
+  let neg = Z.neg
+  let abs = Z.abs
+  let add = Z.add
+  let sub = Z.sub
+  let mul = Z.mul
 
   (* If the first operand of a div is negative, Zarith rounds the result away from zero.
      We thus always transform this into a division with a non-negative first operand.
   *)
-  let div a b = if Big_int_Z.lt_big_int a zero then Big_int_Z.minus_big_int (Big_int_Z.div_big_int (Big_int_Z.minus_big_int a) b) else Big_int_Z.div_big_int a b
+  let div a b = if Z.lt a zero then Z.neg (Z.ediv (Z.neg a) b) else Z.ediv a b
 
-  (* Big_int_Z.mod_big_int computes the Euclidian Modulus, but what we want here is the remainder, as returned by mod on ints
+  (* Z.erem computes the Euclidian Modulus, but what we want here is the remainder, as returned by mod on ints
      -1 rem 5 == -1, whereas -1 Euclid-Mod 5 == 4
   *)
-  let rem a b = Big_int_Z.sub_big_int a (mul b (div a b))
+  let rem a b = Z.sub a (mul b (div a b))
```

Comment:
Is this now just `Z.rem` ?

> Integer remainder. Can raise a Division_by_zero. The result of rem a b has the sign of a, and its absolute value is strictly smaller than the absolute value of b. The result satisfies the equality a = b * div a b + rem a b.

From https://ocaml.org/p/zarith/1.12/doc/Z/index.html#val-rem

Reply:
We have our own `div` definition above, but `Z.div` (as opposed to `Z.ediv`) seems to work the same way, so I'll just remove both.

Reply:
I'll add a qcheck test to make sure they're the same as ours before.